nuxeo-ai-core/src/main/java/org/nuxeo/ai/configuration/ThresholdComponent.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Gethin James
+ *     Andrei Nechaev
+ */
+package org.nuxeo.ai.configuration;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ai.configuration.ThresholdConfiguratorDescriptor.Threshold;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.model.ComponentContext;
+import org.nuxeo.runtime.model.DefaultComponent;
+import org.nuxeo.runtime.model.Descriptor;
+import org.nuxeo.runtime.services.config.ConfigurationService;
+
+/**
+ * Implementation of the ThresholdService
+ */
+public class ThresholdComponent extends DefaultComponent implements ThresholdService {
+
+    private static final Logger log = LogManager.getLogger(ThresholdComponent.class);
+
+    public static final String THRESHOLD_CONFIGURATION_XP = "thresholdConfiguration";
+
+    public static final String DEFAULT_THRESHOLD_VALUE = "nuxeo.ai.default.threshold";
+
+    public static final String AUTOFILL_DEFAULT_VALUE = "nuxeo.ai.autofill.default.threshold";
+
+    public static final String AUTO_CORRECT_DEFAULT_VALUE = "nuxeo.ai.autocorrect.default.threshold";
+
+    protected float globalThreshold = 0.1f;
+
+    protected float globalAutofillThreshold = 0.1f;
```

Comment:
The default should be much higher and settable via nuxeo.conf.  Lets discuss as a team what the levels should be, perhaps 0.85 for autofill and 0.95 for autocorrect?

Reply:
as you said earlier we have lack of description ;) and I guess I need to update readme to include new vars

Reply:
Yes, updating the readme sounds good.